src/uu/ls/src/ls.rs

```diff
@@ -1878,15 +1913,35 @@ fn display_item_long(
             );
         }
 
+        match display_size_or_rdev(md, config) {
+            SizeOrDeviceId::Size(size) => {
+                let _ = write!(out, " {}", pad_left(&size, padding.longest_size_len),);
+            }
+            SizeOrDeviceId::Device(major, minor) => {
+                let _ = write!(
+                    out,
+                    " {}, {}",
+                    pad_left(
+                        &major,
+                        #[cfg(not(unix))]
+                        0usize,
+                        #[cfg(unix)]
+                        padding.longest_major_len,
```

Comment:
Correct me if I'm wrong, because I'm not 100% sure about this, but I think this should be
```rust
padding.longest_major_len.max(padding.longest_size_len - padding.longest_minor_len - 2)
```
to pad the entirety of the device id if a size in the directory is wider than the widest device id. For example:
```
2, 3
123456
```
should be aligned as
```
  2, 3
123456
```

Reply:
Thanks for your review.  I think you may be right.  Good catch!